lib/autoproj/package_manifest.rb

```diff
@@ -216,5 +222,41 @@ def tag_end(name)
                 @tag_text = nil
             end
         end
+
+        # @api private
+        #
+        # REXML stream parser object used to load the XML contents into a
+        # {PackageManifest} object
+        class RosLoader < Loader
+            DEPEND_TAGS = Set['depend', 'build_depend', 'build_export_depend',
+                              'buildtool_depend', 'buildtool_export_depend',
+                              'exec_depend', 'test_depend', 'run_depend']
+
+            def tag_start(name, attributes)
+                if DEPEND_TAGS.include?(name)
+                    @tag_text = ''
+                elsif TEXT_FIELDS.include?(name) || AUTHOR_FIELDS.include?(name)
+                    @tag_text = ''
+                elsif name == 'tags'
+                    @tag_text = ''
+                else
+                    @tag_text = nil
+                end
+            end
+
+            def tag_end(name)
+                if DEPEND_TAGS.include?(name)
+                    manifest.add_dependency(@tag_text)
+                elsif AUTHOR_FIELDS.include?(name)
+                    manifest.send("#{name}s").concat(parse_contact_field(@tag_text))
```

Comment:
package.xml's author field is formatted differently from manifest.xml's (the email is stored as attribute in `package.xml`)

Reply:
Yes, but I autoproj doesn't use/store the email anywhere, does it? That's why I made the conscious decision of ignoring the email. Thoughts?

Reply:
> Yes, but I autoproj doesn't use/store the email anywhere, does it? That's why I made the conscious decision of ignoring the email. Thoughts?

Autoproj does not. Tools layered on top of it does, but then in this case it will simply show no emails (it is supposed to be able to handle that).

Reply:
I could also get the "email" attribute and join that in `@tag_text`. Would you prefer that?

Reply:
> I could also get the "email" attribute and join that in @tag_text. Would you prefer that?

Wouldn't it make more sense to save it and then add the email directly (instead of using `parse_contact_field`) ?

Reply:
Probably, yes. Will do that.